### ISSUE TITLE
Speculation on safely satisfying GDPR/lawyer rich research requests involving any UUID for user_id from Galaxy server tables.

### ISSUE THREAD

**Issue by fubar2** [hexylena/galactic-radio-telescope-client#1]
tl;dr is there a sane way to extend the client to return plots or other non GDPR sensitive artifacts where that is useful?

For the GDPR, tool ids and job ids are ok it seems but any user_id or other identifier like an email are an issue, even if one way hashed AFAIK. So, if you want any tables with UUID or whatever in them, send in the lawyers.

To get around that, a [test research case](https://github.com/fubar2/galcode2data) is available for a situation where you need to return an even less identifiable pdf plot. It's just an MDS plot but it's based on a wide table containing tool use counts for the period with each tool as a column and **each row a user**. The test case script was developed in an IE so could become an interactive tool. Needs to be adjusted with local postgres server creds and run by an admin - with due diligence and other costs. Expensive, but potentially a mechanism to finesse the GDPR for any non-routine queries - but out of scope for this client probably?

For the test case: Easy enough to change it so the plotting is a separate problem, in which case the query and the id hashing can all be done as just another table generator in the client. However! If handing out that data is still a potential GDPR/lawyer thing, even with UUID, it's not a viable solution.
Client's focus is on exporting tables of counts. 

Users doing post processing of tables will want to run plugin R or python code through some kind of interface, with all that extra complexity and need to control risk from running external code - _do not want_ to reinvent a variation on tools for the client!.

**Comment by hexylena** [comment]
Thanks for filing this!
